frontend/src/scenes/insights/insightVizDataLogic.ts

```diff
@@ -348,13 +348,24 @@ export const insightVizDataLogic = kea<insightVizDataLogicType>([
             actions.updateInsightFilter({ display })
         },
         updateQuerySource: ({ querySource }) => {
-            actions.setQuery({
+            interface NodeWithSource {
+                kind: NodeKind
+                source: QuerySourceUpdate
+            }
+
+            const newQuery = {
                 ...values.query,
                 source: {
                     ...values.querySource,
                     ...handleQuerySourceUpdateSideEffects(querySource, values.querySource as InsightQueryNode),
                 },
-            } as Node)
+            } as NodeWithSource
+
+            if (querySource['aggregation_group_type_index'] === undefined) {
+                delete newQuery.source['aggregation_group_type_index']
```

Comment:
hmm, this isn't the right fix, we shouldn't ever need to have any exceptions here, this is the job of `cleanFilters()`, and if not that, all the other helpers in `filterToQueryNode` and `QueryNodeToFilter` applying to `actions.setQuery`

Can we extract this out of this PR? This can be tricky given the complexity here